pyroute2/wirouting.py

```diff
@@ -0,0 +1,72 @@
+""" High-level API built on top of AsyncIPRoute.
+
+This module provides:
+    * Python dataclasses for parsed objects from kernel.
+      You don't need to know NLA attributes names to use it.
+    * Specialized exceptions, so caller does not have to check error code
+      for most commons errors.
+    * Default netlink flags strict_check and ext_ack enabled
+    * Python typing
+"""
+
+import errno
+from typing import Dict, Type
+
+from pyroute2.iproute.linux import AsyncIPRoute
+from pyroute2.netlink.exceptions import NetlinkError
+from pyroute2.netlink.rtnl import (
+    RTM_GETLINK,
+    RTM_NEWLINK,
+    RTM_SETLINK,
+    RTM_NEWLINKPROP,
+)
+
+
+class InterfaceDoesNotExist(NetlinkError):
+    """Requested interface does not exist"""
+
+
+class InterfaceExists(NetlinkError):
+    """Creation failed since interface already exists"""
+
+
+exception_map: Dict[int, Dict[int, Type[Exception]]] = {
+    RTM_NEWLINK: {
+        errno.EEXIST: InterfaceExists,
+        errno.ENODEV: InterfaceDoesNotExist,
+    },
+    RTM_NEWLINKPROP: {errno.ENODEV: InterfaceDoesNotExist},
+    RTM_GETLINK: {errno.ENODEV: InterfaceDoesNotExist},
+    RTM_SETLINK: {
+        errno.EEXIST: InterfaceExists,
+        errno.ENODEV: InterfaceDoesNotExist,
+    },
+}
+
+
+def exception_factory(err, msg):
+    try:
+        return exception_map[msg.orig_type][err.code](*err.args)
+    except LookupError:
+        return err
+
+
+class WiRoute(AsyncIPRoute):
```

Comment:
Do we plan the synchronous counterpart?

Reply:
Not really. It's a good think to provide synchronous code as compatibility for previous code already deployed, but I think that for new code it's better to let caller do loop.run_until_complete()